signature/cose/envelope.go

```diff
@@ -393,15 +394,19 @@ func generateProtectedHeaders(req *signature.SignRequest, protected cose.Protect
 	if !ok {
 		return &signature.InvalidSignRequestError{Msg: "signing scheme: require notary.x509 or notary.x509.signingAuthority"}
 	}
-	protected[signingTimeLabel] = req.SigningTime.Unix()
+	// in go-cose, time.Time will be encoded as numerical (in integer)
+	// representation of seconds since January 1, 1970 UTC.
+	// For more details: https://github.com/fxamacker/cbor/blob/7704fa5efaf3ef4ac35aff38f50f6ff567793072/encode.go#L75
+	// https://github.com/veraison/go-cose/blob/f72b6bc06b90d1e9c07ab11332df92c7838e3027/cbor.go#L30
+	protected[signingTimeLabel] = req.SigningTime
 	if signingTimeLabel == headerLabelAuthenticSigningTime {
 		crit = append(crit, headerLabelAuthenticSigningTime)
 	}
 
 	// expiry
 	if !req.Expiry.IsZero() {
 		crit = append(crit, headerLabelExpiry)
-		protected[headerLabelExpiry] = req.Expiry.Unix()
+		protected[headerLabelExpiry] = req.Expiry
```

Comment:
What's the resulted CBOR object of marshalling a `time.Time`? Do we have a unit test to cover it?

Reply:
> What's the resulted CBOR object of marshalling a `time.Time`? Do we have a unit test to cover it?

As mentioned above, CBOR marshals time.Time to integer. Added unit test to cover this.

Reply:
Code updated per discussion.